File: app/src/main/java/com/example/cykashoppinglist/entity/Item.java
package com.example.cykashoppinglist.entity;

import android.view.View;

import com.example.cykashoppinglist.MainActivity;

import java.util.Date;

public interface Item {
	String getName();
	Date getDate();
	Long getId();
	View.OnClickListener getOnClickListener();
	default String getDateString() {
		return this.getDate() == null ? "" : MainActivity.dateFormat.format(this.getDate());
	}
}

File: app/src/main/java/com/example/cykashoppinglist/entity/Item.kt
package com.example.cykashoppinglist.entity

import android.view.View
import com.example.cykashoppinglist.MainActivity
import java.util.*

interface Item {
	val name: String?
	val date: Date?
	val id: Long?
	fun hasDelete(): Boolean
	val onClickListener: View.OnClickListener?
	val dateString: String?
		get() = if (date == null) "" else MainActivity.dateFormat?.format(date)
}